### PR TITLE
Dapr bot enhancements to create issues based on labels

### PR DESCRIPTION
**Pull request by shubham1172**
Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

# Description

Add a new functionality to the Dapr bot to create issues based on labels.

When an issue is labelled with:
- `docs-needed` - automatically create an issue in dapr/docs
- `sdk-needed` - automatically create issues across SDKs in the dapr org

The authz is implicit since only approvers/maintainers can label issues.

This flow will not run in any of the dapr forks.

## Issue reference

Please reference the issue this PR will close: #5326

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by shubham1172** [comment]
Example issue from `docs-needed` in my fork

image

File: .github/scripts/dapr_bot.js
Comment (ItalyPaleAle) [inline .github/scripts/dapr_bot.js:104]: What if a fork removes this line, would they be able to run this task?

**Review by ItalyPaleAle — APPROVED** [review]
This is cool! Assuming the maintainers don’t have objections, LGTM!

File: .github/scripts/dapr_bot.js
Comment (berndverst) [reply .github/scripts/dapr_bot.js:104]: They don't have access to the bot credentials.. we do not have credential sharing enabled. So no, they can't really run this.

**Review by berndverst — APPROVED** [review]

**Comment by berndverst** [comment]
> This is cool! Assuming the maintainers don’t have objections, LGTM!

This is something we discussed with maintainers quite some time ago. Maybe ~3 months? :)

I'm sure @artursouza or @yaron2 remember.

**Comment by berndverst** [comment]
@shubham1172 can you do be a favor an open this same PR in Contrib repo? But not the SDK part -- just the docs part. The label to use there is `documentation required`. That would be super helpful, thanks!

**Comment by shubham1172** [comment]
> @shubham1172 can you do be a favor an open this same PR in Contrib repo? But not the SDK part -- just the docs part. The label to use there is `documentation required`. That would be super helpful, thanks!

Yes sure, I will do that. Well https://github.com/dapr/dapr/issues/3926 will help us not duplicate code, but for now I will just create a PR in contrib to add this functionality.

**Review by artursouza — APPROVED** [review]

**Pull request merged**
